cmake/cable/buildinfo/buildinfo.c.in

```diff
@@ -7,7 +7,7 @@
 
 #include "buildinfo.h"
 
-const struct buildinfo* @FUNCTION_NAME@()
+const struct buildinfo* @FUNCTION_NAME@(void)
```

Comment:
Why are these needed?

Reply:
```
/evmone/build/test/t8n/evmone_t8n/buildinfo.h:29:49: error: a function declaration without a prototype is deprecated in all versions of C [-Werror,-Wstrict-prototypes]
const struct buildinfo* evmone_t8n_get_buildinfo();
                                                ^
                                                 void
```

Reply:
Yeah but why wasn't this a problem before?

Reply:
It's enabled by default since clang 15. 
https://releases.llvm.org/14.0.0/tools/clang/docs/DiagnosticsReference.html#wstrict-prototypes
https://releases.llvm.org/15.0.0/tools/clang/docs/DiagnosticsReference.html#wstrict-prototypes

Reply:
Yeah but why is `master` working?

Reply:
Because ci uses old clang version probably.

Reply:
And this lib has not been used before

Reply:
Are you sure you need `(void)` in the function definition?

FYI this is related to C23 where the meaning of declaration of `int f()` changes from `int f(...)` to `int f(void)`. But I think they overdone the warning.

Reply:
So for `clang 16.0.0` I need but for `appleclang 14.0.0.14000029`I don't need. I can stop using newer compiler and we don't need to change anything.

Reply:
but it looks that it's not only my problem ;) https://app.circleci.com/pipelines/github/ethereum/evmone/4335/workflows/d5b6d7cf-9826-45b1-8586-a9366b89a908/jobs/47373

Reply:
I think i testes definition too, but will check again on CI

Reply:
Also the same error.
https://app.circleci.com/pipelines/github/ethereum/evmone/4336/workflows/2a430d17-fbc4-428c-ac0e-80aa38298ef5/jobs/47384